extensions/notebook/src/dialog/managePackages/addNewPackageTab.ts

```diff
@@ -0,0 +1,264 @@
+				.withFormItems([{
+					component: this.newPackagesSearchBar,
+					title: ''
+				}, {
+					component: this.packagesSearchButton,
+					title: ''
+				}, {
+					component: this.newPackagesNameLoader,
+					title: this.PackageNameTitle
+				}, {
+					component: this.newPackagesVersionsLoader,
+					title: this.PackageVersionTitle
+				}, {
+					component: this.newPackagesSummaryLoader,
+					title: this.PackageSummaryTitle
+				}, {
+					component: this.packageInstallButton,
+					title: ''
+				}]).component();
+
+			await view.initializeModel(formModel);
+		});
+	}
+
+	public get tab(): azdata.window.DialogTab {
+		return this.addNewPkgTab;
+	}
+
+	private async toggleNewPackagesFields(enable: boolean): Promise<void> {
+		await this.packagesSearchButton.updateProperties({ enabled: enable });
+		await this.newPackagesNameLoader.updateProperties({ loading: !enable });
+		await this.newPackagesVersionsLoader.updateProperties({ loading: !enable });
+		await this.newPackagesSummaryLoader.updateProperties({ loading: !enable });
+	}
+
+	private async loadNewPackageInfo(): Promise<void> {
+		await this.packageInstallButton.updateProperties({ enabled: false });
+		await this.toggleNewPackagesFields(false);
+		try {
+			let packageName = this.newPackagesSearchBar.value;
+			if (!packageName || packageName.length === 0) {
+				return;
+			}
+
+			let pipPackage = await this.fetchPypiPackage(packageName);
+			if (!pipPackage.versions || pipPackage.versions.length === 0) {
+				this.dialog.showErrorMessage(this.NoVersionsFoundError);
+				return;
+			}
+
+			await this.newPackagesName.updateProperties({
+				value: packageName
+			});
+			await this.newPackagesVersions.updateProperties({
+				values: pipPackage.versions,
+				value: pipPackage.versions[0],
+			});
+			await this.newPackagesSummary.updateProperties({
+				value: pipPackage.summary
+			});
+
+			// Only re-enable install on success
+			await this.packageInstallButton.updateProperties({ enabled: true });
+		} catch (err) {
+			this.dialog.showErrorMessage(utils.getErrorMessage(err));
+
+			await this.newPackagesName.updateProperties({
+				value: this.InvalidTextPlaceholder
+			});
+			await this.newPackagesVersions.updateProperties({
+				value: this.InvalidTextPlaceholder,
+				values: [this.InvalidTextPlaceholder],
+			});
+			await this.newPackagesSummary.updateProperties({
+				value: this.InvalidTextPlaceholder
+			});
+		} finally {
+			await this.toggleNewPackagesFields(true);
+		}
+	}
+
+	private async fetchPypiPackage(packageName: string): Promise<PipPackageData> {
+		return new Promise<PipPackageData>((resolve, reject) => {
+			request.get(`https://pypi.org/pypi/${packageName}/json`, { timeout: 10000 }, (error, response, body) => {
+				if (error) {
+					return reject(error);
+				}
+
+				if (response.statusCode === 404) {
+					return reject(this.PackageNotFoundError);
+				}
+
+				if (response.statusCode !== 200) {
+					return reject(response.statusMessage);
```

Comment:
Is this wrapped upstream? E.g. will we just give a `FORBIDDEN` error message if we get a 401, or would we have a user-friendly message to wrap it?